File: practice/LinkedList.java
public class LinkedList {
    private class Node {
        public int data;
        public Node next;

        public Node(int data) {
            this.data = data;
        }
    }

    private Node root;

    public void add(int data) {
        if (root == null) {
            root = new Node(data);
            return;
        } else {
            Node tmp = new Node(data);
            tmp.next = root;
            root = tmp;
        }
    }

    public int get(int index) {
        int i = 0;
        Node tmp = root;
        while (tmp.next != null && i < index) {
            i++;
            tmp = root.next;
        }
        return root.data;
    }

    public static void main(String[] args) {

    }
}

File: practice/linkedlist.cpp
#include <bits/stdc++.h>

using namespace std;

class Node {
public:
    int data;
    Node *next;
    Node(int data) {
        this->data = data;
        this->next = nullptr;
    }
};

Node *head;

void insert(int data) {
    if (head == nullptr) {
        head = new Node(data);
    } else {
        Node *tmp = new Node(data);
        tmp->next = head;
        head = tmp;
    }
}

void printSolution() {
    Node *tmp = head;
    while (tmp != nullptr) {
        cout << tmp->data << endl;
        tmp = tmp->next;
    }
}

int main() {
    for (int i = 1; i <= 10; i++) {
        insert(i);
    }
    printSolution();
}